snapcraft/cli/templates.py

```diff
@@ -0,0 +1,68 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import os
+import yaml
+import sys
+
+from ._options import get_project
+from snapcraft.internal import common, project_loader
+
+
+@click.group()
+def templatecli(**kwargs):
+    pass
+
+
+@templatecli.command()
+def templates(**kwargs):
+    """List available templates."""
+
+    template_names = os.listdir(common.get_templatesdir())
+    if not template_names:
+        click.echo("No templates supported")
+        return
+
+    # Wrap the output depending on terminal size
+    width = common.get_terminal_width()
+    for line in common.format_output_in_columns(
+        sorted(template_names), max_width=width
```

Comment:
alternatively we should list what bases it supports here

Reply:
I agree, let me try some different formatting, here.